rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -2064,19 +2069,41 @@ extern void ppu_finalize(const ppu_module& info)
 #endif
 }
 
-extern void ppu_precompile(std::vector<std::string>& dir_queue)
+extern void ppu_precompile(std::vector<std::string>& dir_queue, std::vector<lv2_prx*>* loaded_prx)
 {
-	std::vector<std::pair<std::string, u64>> file_queue;
-	file_queue.reserve(2000);
+	// Remove duplicates
+	std::sort(dir_queue.begin(), dir_queue.end());
+	dir_queue.erase(std::unique(dir_queue.begin(), dir_queue.end()), dir_queue.end());
 
-	// Initialize progress dialog
-	g_progr = "Scanning directories for SPRX libraries...";
+	const std::string firmware_sprx_path = vfs::get("/dev_flash/sys/external/");
+
+	// Map fixed address executables area, fake overlay support
+	const bool had_ovl = !vm::map(0x3000'0000, 0x1000'0000, 0x200).operator bool();
+	const u32 ppc_seg = std::exchange(g_ps3_process_info.ppc_seg, 0x3);
+
+	enum class exec_type
+	{
+		none,
+		prx,
+		ovl,
+		//elf,
+	};
+
+	std::vector<std::tuple<std::string, exec_type, u64>> file_queue;
+	file_queue.reserve(2000);
 
 	// Find all .sprx files recursively (TODO: process .mself files)
-	for (usz i = 0; i < dir_queue.size(); i++)
+	for (usz i = 0;; i++)
 	{
+		if (i >= dir_queue.size())
+		{
+			g_progr_ftotal = file_queue.size();
```

Comment:
I don't think it's right to count 1 MSELF as 1 file. Because technically it does a very little difference, whether we compile from mself or from bare file.

Reply:
Nvm, the check is rather strange.

Reply:
It doesn't count as 1 file I think, there's an entry in file_queue for each record right?